src/ng2-smart-table/components/cell/edit-cell.component.ts

```diff
@@ -10,9 +10,31 @@ export class EditCellComponent {
 
   @Input() cell: Cell;
   @Input() inputClass: string = '';
+  @ViewChild('dynamicTarget', {read: ViewContainerRef}) dynamicTarget: any;
+  customComponent: any;
 
   @Output() public edited: EventEmitter<any> = new EventEmitter<any>();
 
+  constructor(private resolver: ComponentFactoryResolver){
+  }
+
+  ngOnChanges(arg) {
```

Comment:
In my view, edit-cell is just a container which includes one or another editor component. But, any user-created custom editor needs some additional logic to be initiated, so it makes sense to create a separate CustomEditorComponent which will instantiate any custom user-provided component.